src/graph/test/modifiers.ts

```diff
@@ -0,0 +1,1062 @@
+import t from 'tap'
+import {
+  asGraphModifiersConfig,
+  assertGraphModifiersConfig,
+  GraphModifier,
+} from '../src/modifiers.ts'
+import { Spec } from '@vltpkg/spec'
+import type { SpecOptions } from '@vltpkg/spec'
+import { reload } from '@vltpkg/vlt-json'
+import type { Node } from '../src/node.ts'
+import { Edge } from '../src/edge.ts'
+import type { GraphLike, NodeLike, EdgeLike } from '../src/types.ts'
+import type { DependencyTypeShort, Manifest } from '@vltpkg/types'
+import { joinDepIDTuple } from '@vltpkg/dep-id'
+
+// Graph fixtures code
+const specOptions: SpecOptions = {
+  registries: {
+    custom: 'http://example.com',
+  },
+}
+
+const projectRoot = '.'
+
+const newGraph = (rootName: string): GraphLike => {
+  const graph = {} as GraphLike
+  const addNode = newNode(graph)
+  const mainImporter = addNode(rootName)
+  mainImporter.id = joinDepIDTuple(['file', '.'])
+  mainImporter.mainImporter = true
+  mainImporter.importer = true
+  mainImporter.graph = graph
+  graph.importers = new Set([mainImporter])
+  graph.mainImporter = mainImporter
+  graph.nodes = new Map([[mainImporter.id, mainImporter]])
+  graph.edges = new Set()
+
+  return graph
+}
+
+const newNode =
+  (graph: GraphLike) =>
+  (name: string, version = '1.0.0'): NodeLike => ({
+    projectRoot,
+    confused: false,
+    edgesIn: new Set(),
+    edgesOut: new Map(),
+    importer: false,
+    mainImporter: false,
+    graph,
+    id: joinDepIDTuple(['registry', '', `${name}@${version}`]),
+    name,
+    version,
+    location:
+      'node_modules/.vlt/;;${name}@${version}/node_modules/${name}',
```

Comment:
This should be `node_modules/.vlt/${joinDepIdTuple(['registry', '', name + '@' + version])}/node_modules/${name}`.

We don't use `;` anymore, because it breaks node-gyp.